url913d8566d1_AnsBalin_rbn_tgz-pJ8-MainJ8/src/DataOutput.java

```diff
@@ -0,0 +1,98 @@
+import java.io.*;
+import java.util.*;
+
+public class DataOutput{
+
+  private String path;
+  private boolean append = false;
+
+  public DataOutput(String path){
+    
+    this.path = path;
+    
+  }
+  
+  public DataOutput(String path, boolean append){
+  
+    this.path = path;
+    this.append = append;
+    
+  }
+  
+  public String getPath(){
+    
+    return path;
+    
+  }
+  
+  public void writeToFile(int[] arr) throws IOException{
+  
+    FileWriter write = new FileWriter(path, append);
+    PrintWriter print_line = new PrintWriter(write);
+    
+    for(int i=0; i<arr.length; i++){
+      print_line.printf("%d\t", arr[i]);
+    }
+    print_line.printf("\n");
+
+    print_line.close();
+  
+  }
+  
+  public void writeToFile(double[] arr) throws IOException{
+    
+    FileWriter write = new FileWriter(path, append);
+    PrintWriter print_line = new PrintWriter(write);
+    
+    for(int i=0; i<arr.length; i++){
+      print_line.printf("%f\t", arr[i]);
+    }
+    print_line.printf("\n");
+    
+    print_line.close();
+    
+  }
+  
+  public void writeToFile(String str) throws IOException{
+  
+    FileWriter write = new FileWriter(path, append);
+    PrintWriter print_line = new PrintWriter(write);
+    
+    print_line.printf("%s\n", str);
+    print_line.close();
+  
+  }
+  
+  public void writeToFile(Object obj) throws IOException{
+  
+    FileOutputStream fos = new FileOutputStream(path, append);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *fos*. The resource is referenced at lines 69, 71, 73. The resource is closed at line 74. There are other execution paths that do not contain closure statements, for example, when ObjectOutputStream.writeObject throws an exception. Ensure *fos* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).